### PR TITLE
CAMEL-15409: ArangoDb - add operations on graphs in the producer

### PR DESCRIPTION
**Pull request by zbendhiba**
- This is PR where I've added operations on graphs + tests.

- I've also changed in the producer how Camel performs the AQL query, to return a java collection and not the Cursor object from Arango java driver.

- I did some refactoring on tests.

- I've updated the first version of the component : 3.5.0 instead of 3.5.0-SNAPSHOT

### CODE REVIEW

File: components/camel-arangodb/src/generated/resources/org/apache/camel/component/arangodb/arangodb.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
     "title": "ArangoDb",
     "description": "Perform operations on ArangoDb documents, collections and graphs.",
     "deprecated": false,
-    "firstVersion": "3.5.0-SNAPSHOT",
+    "firstVersion": "3.5.0",
```
[the remark this answers is not in the document]
Comment (zbendhiba) [reply components/camel-arangodb/src/generated/resources/org/apache/camel/component/arangodb/arangodb.json:8]: So do I drop that change?
Manually I did the change only on the endpoint class. Then with mvn install it changes the first version everywhere.
I noticed it's the only one with snapshot in first version

**Comment by davsclaus** [comment]
Using the version with -SNAPSHOT is correct when you set it in the UriEndpoint annotation.

**Comment by davsclaus** [comment]
Would it be good to update the documentation (eg src/main/docs) about the graph operation.

**Comment by zbendhiba** [comment]
I set back the first version to SNAPSHOT.
I've added some descriptions for the attributes. @davsclaus  is it ok ?

**Comment by davsclaus** [comment]
Oh firstVersion should be 3.5.0.
Remove the SNAPSHOT there.

Its only in the maven pom.xml files etc that the maven release plugin can change the version numbers.

**Comment by zbendhiba** [comment]
done !!

**Pull request merged**